enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/DebouncedColorPillPicker.tsx

```diff
@@ -0,0 +1,32 @@
+import { useDebouncedCallback } from "@mantine/hooks";
+import { useState } from "react";
+
+import { ColorPillPicker } from "metabase/common/components/ColorPicker";
+
+export const DebouncedColorPillPicker = ({
+  initialValue,
+  onChange,
+  debounceMs,
+}: {
+  initialValue: string;
+  onChange: (value: string) => void;
+  debounceMs: number;
+}) => {
+  const [previewValue, setPreviewValue] = useState<string>(initialValue);
+
+  const debouncedUpdate = useDebouncedCallback(onChange, debounceMs);
```

Comment:
This makes the color picking experience smoother, as updating the embed settings directly when dragging on the color picker causes performance hits.

Reply:
> TBH, I think this functionality should be baked into the main component itself. Having to debounce on call sites doesn't look great.

I was thinking the same as well, it just feels a little weird that `ColorPillPicker` is the only component that would have baked-in debounces while it's siblings (e.g. `ColorPicker.tsx`) does not have one. I think I'll go ahead and bake it in anyway.

> Secondly, I wouldn't expect the callsite to have to provide debounceMs every time. Given this component could be used in multiple places. Or at least have some default value so that we don't need to specify debounceMs every time.

Good idea, lemme add a default.